app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -122,6 +121,28 @@ def bulk_resume
         }
       end
 
+      api :POST, '/tasks/bulk_cancel', N_('Cancel all cancellable tasks')
+      param :search, String, :desc => N_('Cancel tasks matching search string')
+      param :task_ids, Array, :desc => N_('Cancel specific tasks by ID')
+      def bulk_cancel
+        if params[:search].nil? && params[:task_ids].nil?
+          raise ArgumentError, _('Please provide at least one of search or task_ids parameters in the request')
+        end
+        scope = resource_scope
+        scope = scope.search_for(params[:search]) if params[:search]
+        scope = scope.select('DISTINCT foreman_tasks_tasks.*')
+        scope = scope.where(:id => params[:task_ids]) if params[:task_ids]
```

Comment:
Maybe this could be extracted into common `bulk_scope` helper which could be used here and in `bulk_resume`?

Reply:
added the `bulk_scope` here and in the first PR too:
https://github.com/theforeman/foreman-tasks/pull/516